edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -64,8 +64,7 @@ public class EdgeAgentConnection : IEdgeAgentConnection
             this.reportedProperties = Option.None<TwinCollection>();
             this.deviceClient = Option.None<IModuleClient>();
             this.retryStrategy = Preconditions.CheckNotNull(retryStrategy, nameof(retryStrategy));
-            this.refreshTimer = new Timer(refreshConfigFrequency.TotalMilliseconds);
-            this.refreshTimer.Elapsed += (_, __) => this.RefreshTimerElapsed();
+            this.refreshTwinTask = new PeriodicTask(this.RefreshTwinAsync, refreshConfigFrequency, refreshConfigFrequency, Events.Log, "refresh twin config");
```

Comment:
this.RefreshTwinAsync can be called from this PeriodicTask, CreateAndInitDeviceClient, OnConnectionStatusChanged, and OnDesiredPropertiesUpdated.  First periodic task doesn't call twinLock.LockAsync before calling RefreshTwinAsync; may move twinLock.LockAsync statement inside RefreshTwinAsync method.

Besides that I concern this may run too often in some cases, for example if it triggered by OnDesiredPropertiesUpdated, and then periodictask also trigger right after;  I would suggest to reset timer inside periodictask after RefreshTwinAsync called.

Reply:
yes, we had the reset logic earlier... but it is non-trivial to do it with a loop and I wanted to avoid the complexity that comes with it. This also makes it easier to track logs to see if the periodic task is working properly.